sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -25,69 +31,165 @@ template <typename T, access::address_space Space> class multi_ptr;
 
 namespace detail {
 
-template <typename> struct is_vec : std::false_type {};
-template <typename T, std::size_t N>
-struct is_vec<cl::sycl::vec<T, N>> : std::true_type {};
+namespace sub_group {
 
-template <typename T, __spv::GroupOperation O>
-static typename std::enable_if<
-    !detail::is_floating_point<T>::value && std::is_signed<T>::value, T>::type
-calc(T x, intel::minimum<T> op) {
-  return __spirv_GroupSMin(__spv::Scope::Subgroup, O, x);
+template <typename T> T broadcast(T x, id<1> local_id) {
+  using OCLT = detail::ConvertToOpenCLType_t<T>;
+  OCLT Arg = detail::convertDataToType<T, OCLT>(x);
+  OCLT Ret =
+      __spirv_GroupBroadcast(__spv::Scope::Subgroup, Arg, local_id.get(0));
+  return detail::convertDataToType<OCLT, T>(Ret);
 }
 
-template <typename T, __spv::GroupOperation O>
-static typename std::enable_if<
-    !detail::is_floating_point<T>::value && std::is_unsigned<T>::value, T>::type
-calc(T x, intel::minimum<T> op) {
-  return __spirv_GroupUMin(__spv::Scope::Subgroup, O, x);
-}
+#define GENERATE_BODY_1ARG(name, SPIRVOperation)                               \
```

Comment:
You should have for all your macro some more protected names agains conflicts, such as with a `__SYCL_` prefix

Reply:
Done